src/resource/solidDataset.ts

```diff
@@ -233,11 +233,19 @@ export async function responseToSolidDataset(
           solidDataset = addRdfJsQuadToDataset(solidDataset, quad);
         }
       });
-      parser.onComplete(() => {
+      parser.onComplete(async () => {
+        // If a Resource contains more than this number of Blank Nodes,
+        // we consider the detection of chains (O(n^2), I think) to be too
+        // expensive, and just incorporate them as regular Blank Nodes with
+        // non-deterministic, ad-hoc identifiers into the SolidDataset:
+        const maxBlankNodesToDetectChainsFor = 20;
```

Comment:
Should this be a constant in the `constants.ts` file ? So far it's only RDF constants, but it feels weird to have a hard-coded limit defined so locally.

Reply:
Not sure, since it's really only used locally, i.e. there's just a single mention of this variable - I mostly extracted it to be able to give it a name that explained its purpose, rather than for it to be a magic number. I'm afraid that pulling it out of its context will make it _harder_ to understand what it does, and it's not meant to be re-used elsewhere in the codebase either.